### ISSUE TITLE
Best method for feature selection

### ISSUE THREAD

**Issue by Aniruddha25** [rasbt/mlxtend#429]
I am trying to combine yellowbrick's RFECV and mlxtend's Exhaustive feature selection to find best subset 
of features.In code i substitute min_features parameter in EFS =optimal no. of features by yellowbrick's RFECV.I have applied this on boston dataset.is this valid approach? and overall what is the best method for 
feature selection for regression problems ?
Code-
from sklearn.neighbors import KNeighborsRegressor
from mlxtend.feature_selection import ExhaustiveFeatureSelector as EFS
from sklearn.datasets import load_boston
from sklearn.linear_model import Ridge
from sklearn.linear_model import RANSACRegressor
from sklearn.linear_model import LinearRegression
from yellowbrick.features import RFECV
from sklearn.ensemble import GradientBoostingRegressor
boston = load_boston()
X, y = boston['data'], boston['target']
model=Ridge()
print('Optimum no. of features to be selected using Recursive feature elimination')   
viz = RFECV(model)
viz.fit(X, y)
k=int(viz.n_features_)
print(k)
viz.poof()
efs1 = EFS(model, 
           min_features=k,
           max_features=12,
           scoring='r2',
           cv=14)

efs1 = efs1.fit(X, y)

print('Best accuracy score: %.2f' % efs1.best_score_)
print('Best subset (indices):', efs1.best_idx_)
Output-
<img width="755" alt="mlxtend-yellowbricks" src="https://user-images.githubusercontent.com/41265081/44506554-82058200-a6c4-11e8-8efb-7231e6a62f93.PNG">

**Comment by rasbt** [comment]
It's hard to say what the best method for feature selection is on a regression problem. I would say that exhaustive feature selection will guarantee to give you optimal results since you are sampling all possible combinations. But it is expensive. Recursive Feature Elimination may be the computationally cheapest approach, because it is selecting based on coefficients. This does only make sense though if you have a linear regression problem. SequentialFeatureSelection is more of a universal method that doesn't make any assumptions and is a heuristic for exhaustive feature selection.

Hope that helps!

**Comment by Aniruddha25** [comment]
Thank you .
